boards/raspberrypi/rpi_pico2/rpi_pico2-pinctrl.dtsi

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, Andrew Featherstone
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/dt-bindings/pinctrl/rpi-pico-rp2040-pinctrl.h>
```

Comment:
![pinfunc](https://github.com/user-attachments/assets/0d0babb1-04b2-42d1-925b-f708c0b1c3d0)

Pin Function definitions are different between rp2040 and rp2350.
(For example, USB moved from FUNC9 to FUNC10.)
So, we need to create `rpi-pico-rp2040-pinctrl.h`.

Reply:
@soburi please don't feel the need to take screenshots of the datasheet. 

I am aware of the differences here, I even have my own local changes to support this, but at the point I submitted the PR there was no need for this.

Reply:
I can accept if some part of lacks, but this is wrong and needs to be fixed.